packages/realm-server/server.ts

```diff
@@ -601,6 +603,51 @@ export class RealmServer {
 
     throw new Error(`Can not determine the matrix registration secret`);
   }
+
+  private fetchCatalogRealms(): (
+    ctxt: Koa.Context,
+    _next: Koa.Next,
+  ) => Promise<any> {
+    return async (ctxt: Koa.Context, _next: Koa.Next) => {
+      let results = await fetchPublicRealms(this.dbAdapter);
+      let promises = results.map(async ({ realm_url: realmURL }) => {
```

Comment:
lets rename `promises` to `publicRealmPromises`. optionally you could just wrap line 642's `Promise.all` around this and not even need a `promises` variable